R/lmer_multimember.R

```diff
@@ -132,17 +138,29 @@ lmer <- function(formula,
     control = control$optCtrl
   )
 
-  m1 <- mkMerMod(environment(devfun),
+  res <- mkMerMod(
+    environment(devfun),
     opt,
     lmod$reTrms,
     fr = lmod$fr,
     mc = orig_call
   )
 
-  # convert model object to lmerModMultiMember
-  m1 <- as(m1, "lmerModMultiMember")
-  m1@memberships <- memberships
-  return(m1)
+  if (lmlib == "lmerTest") {
```

Comment:
R supports objects belonging to multiple classes, right? I would be tempted to just have the `lmerModLmerTest` be an extra class along with `lmerMod` and `lmerModMultiMember`.

Reply:
Seems like that would be less messy, yes. I'm 100% not sure how I'd need to set up the multiple dispatch for `summary` and `print.summary`, because those need to be handled differently depending on whether the model is `lmerTest`, `lmerMultiMember`, or both.

Reply:
This doesn't work, by the way. There's multiple inheritance for S4 objects, but not multiple class membership. So `lmerModMultiMember` could, in principle, inherit from `lmerModLmerTest`, but that would break if `lmerTest` is not installed. I can maybe rig up an optional inheritance scheme (e.g. `lmerModMultiMember` only inherits from `lmerModLmerTest` if `lmerTest` is installed) but I'm not sure if having a model object _that sometimes inherits a class from another package and sometimes doesn't_ is any less messy than what we have now.

Will give this some thought. It's possible that with some rethinking none of this mess is necessary at all.